### PR TITLE
Fix compilation of hackage-security test suite with Cabal-syntax on Hackage and bump version of hackage-repo-tool before release

### PR DESCRIPTION
**Pull request by Mikolaj**

### CODE REVIEW

**Comment by Mikolaj** [comment]
See also https://github.com/haskell/cabal/issues/8371 and https://github.com/haskell/cabal/issues/8370.

**Review by ulysses4ever — APPROVED** [review]

File: hackage-security/hackage-security.cabal [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cabal-version:       1.12
 name:                hackage-security
-version:             0.6.2.1
+version:             0.6.2.2
```
Comment (andreasabel) [inline hackage-security/hackage-security.cabal:3]: If you bump the version, you should delete the `x-revision` field, (or set the revision to `0` I suppose, but I never tried this).

**Review by andreasabel — CHANGES_REQUESTED** [review]

File: hackage-security/hackage-security.cabal (hunk 1 above)
Comment (Mikolaj) [reply hackage-security/hackage-security.cabal:3]: Done. Thank you.

**Review by andreasabel — APPROVED** [review]
Thanks!

**Review by Kleidukos — APPROVED** [review]

**Comment by Mikolaj** [comment]
I wasn't able to revise on Hackage the Cabal-syntax breakage that requires the last commit, so I'm merging with it included, as a workaround.

File: hackage-security/hackage-security.cabal (hunk 1 above)
Comment (gbaz) [reply hackage-security/hackage-security.cabal:3]: its not clear to me why there's an x-revision field in the cabal file in the repo at all? typically x-revision is supposed to be managed for you by hackage server.

Comment (fgaz) [reply hackage-security/hackage-security.cabal:3]: Makes it easier to automatically upload the revision with https://github.com/hackage-trustees/hackage-cli

**Pull request merged**